tools/api-markdown-documenter/src/LintApiModel.ts

```diff
@@ -0,0 +1,308 @@
+	 * The name of the package that the {@link ReferenceError.sourceItem} belongs to.
+	 */
+	readonly packageName: string;
+}
+
+/**
+ * An error resulting from an `{@inheritDoc}` tag with an invalid target.
+ */
+// eslint-disable-next-line unicorn/prevent-abbreviations
+export interface InheritDocReferenceError
+	extends LinterErrorBase<LinterErrorKind.InheritDocReferenceError>,
+		ReferenceError {
+	/**
+	 * {@inheritDoc LinterErrorBase.kind}
+	 */
+	readonly kind: LinterErrorKind.InheritDocReferenceError;
+}
+
+/**
+ * An error resulting from a `{@link}` tag with an invalid target.
+ */
+export interface LinkReferenceError
+	extends LinterErrorBase<LinterErrorKind.LinkReferenceError>,
+		ReferenceError {
+	/**
+	 * {@inheritDoc LinterErrorBase.kind}
+	 */
+	readonly kind: LinterErrorKind.LinkReferenceError;
+}
+
+/**
+ * {@link LinterResult} base type.
+ */
+export interface LinterResultBase<TSuccess extends boolean> {
+	/**
+	 * Result "success" status.
+	 */
+	readonly success: TSuccess;
+}
+
+/**
+ * A successful linter result. No issues were found.
+ */
+export interface LinterSuccessResult extends LinterResultBase<true> {
+	/**
+	 * {@inheritDoc LinterResultBase.status}
+	 */
+	readonly success: true;
+}
+
+/**
+ * Success result singleton.
+ */
+const successResult: LinterSuccessResult = { success: true };
+
+/**
+ * A linter failure result. One or more issues were found.
+ */
+export interface LinterFailureResult extends LinterResultBase<false> {
+	/**
+	 * {@inheritDoc LinterResultBase.status}
+	 */
+	readonly success: false;
+
+	/**
+	 * `{@inheritDoc}` reference errors found in the API model.
+	 */
+	readonly inheritDocReferenceErrors: readonly InheritDocReferenceError[];
+
+	/**
+	 * `{@link}` reference errors found in the API model.
+	 */
+	readonly linkReferenceErrors: readonly LinkReferenceError[];
+}
+
+/**
+ * Result of {@link lintApiModel}.
+ * @remarks Includes a {@link LinterResult.success} indicating whether or not the check was a "success" (no issues found) or "failure" (issues found).
+ */
+export type LinterResult = LinterSuccessResult | LinterFailureResult;
+
+/**
+ * Validates the given API model.
+ *
+ * @returns A "Result" object indicating whether or not the check was a "success" (no issues found) or "failure" (issues found).
+ * In the case of a "failure", all issues found are included.
+ *
+ * @throws
+ * If the specified {@link LoadModelOptions.modelDirectoryPath} doesn't exist, or if no `.api.json` files are found directly under it.
+ */
+export async function lintApiModel(options: LintApiModelOptions): Promise<LinterResult> {
+	const optionsWithDefaults: Required<LintApiModelOptions> = {
+		...defaultLintApiModelOptions,
+		...options,
+	};
+	const { modelDirectoryPath, logger } = optionsWithDefaults;
+
+	logger.verbose("Loading API model...");
+
+	// Load the model
+	// Use a no-op logger to prevent logging during the load process
+	const apiModel = await loadModel({ modelDirectoryPath, logger: noopLogger });
```

Comment:
I imagine we already load the model in some other place... if so, should we think about passing it already loaded to `lintApiModel` instead of loading it again? Not particularly perf-sensitive code, but to me it makes sense conceptually to do it that way. Unless there's a particular reason not to.

Reply:
Yeah, that probably makes sense. We may want both kinds of APIs when we eventually expose this, but until the desired shape is better understood, keeping this simpler probably makes more sense. Will update.